src/lib/src/service/execution/test.execution.service.ts

```diff
@@ -24,9 +26,23 @@ export class TestExecutionService {
     return this.http.get(this.getURL(path, TestExecutionService.statusURLPath) + '&wait=true').toPromise();
   }
 
+  statusAll(): Promise<Map<string, ElementState>> {
```

Comment:
'statusAll' is a bit misleading here, since not all test status are returned, only the ones the server cares to provide (afaik). I can't think of a better name right now, but maybe you have an idea?

Reply:
Well, the server is supposed to return the state of all tests it knows about, so that's the notion of 'all' here. Its basically the status of all tests except those that have never been executed, which, one could argue, don't really have a status. That's pretty much how it is implemented in the backend, while the frontend uses `IDLE` as the state for anything it doesn't know better, i.e. it is that state of tests that have never been run, but also the state of non-executable files and folders…

Long story short, I can't really think of a concise, yet more descriptive name here, and I don't think its all that misleading.
Maybe we [c|sh]ould add some documentation to our interfaces / REST endpoints – here, and in general – so the intended contract / semantics are made more explicit(?)